experimental/Schemes/IdealSheaves.jl

```diff
@@ -149,43 +149,33 @@ ideal_sheaf(X::AbsCoveredScheme) = IdealSheaf(X)
 # set up an ideal sheaf by automatic extension
 # from one prescribed set of generators on one affine patch
 @doc raw"""
-    IdealSheaf(X::AbsCoveredScheme, U::AbsAffineScheme, g::Vector)
+    IdealSheaf(X::AbsCoveredScheme, U::AbsAffineScheme, I::Ideal)
 
-Set up an ideal sheaf on ``X`` by specifying a set of generators ``g``
+Set up an ideal sheaf on ``X`` by specifying an ideal ``I``
 on one affine open subset ``U`` among the `basic_patches` of the
 `default_covering` of ``X``.
 
-**Note:** The set ``U`` has to be dense in its connected component
-of ``X`` since otherwise, the extension of the ideal sheaf to other
-charts can not be inferred.
+**Note:** The ideal ``I`` has to be prime so that its extension 
+to the other charts of ``X`` is well defined. 
 """
 function IdealSheaf(
     X::AbsCoveredScheme, U::AbsAffineScheme, 
-    g::Vector{RET}; check::Bool=false
-  ) where {RET<:RingElem}
-  I = ideal(OO(U), g)
+    I::Ideal; check::Bool=false
+  )
+  @assert base_ring(I) === OO(U) || error("ideal not defined in the correct ring")
   @check is_prime(I) "ideal must be prime"
   return PrimeIdealSheafFromChart(X, U, I)
+end
```

Comment:
```suggestion
end

function IdealSheaf(X::AbsAffineScheme, I::Ideal; covered_scheme::AbsCoveredScheme=CoveredScheme(X))
  @assert base_ring(I) === OO(X)
  @assert length(affine_charts(covered_scheme)) == 1 && X === first(affine_charts(covered_scheme))
  return IdealSheaf(covered_scheme, IdDict{AbsAffineScheme, Ideal}([X=>I]); check=false)
end
```

@afkafkafk13 : I propose to add and use this convenience method for your purposes. Also with a view towards the failing tests.

Reply:
Thank you, This is a very convenient compatibility function.